fragment_info.go

```diff
@@ -570,6 +570,17 @@ func (fI *FragmentInfo) DumpSTDOUT() error {
 	return nil
 }
 
+// String retrieves the string representation of the FragmentInfo
+func (fI *FragmentInfo) String() (string, error) {
```

Comment:
Same with returning `error` here. Returning just a string would enable `FragmentInfo` to participate in [string formatting](https://pkg.go.dev/fmt). We shouldn't panic in this case, but return the error string or an `<ERROR>` placeholder.

Reply:
Done, thanks for the review!